codalab/lib/beam/MultiReaderFileStream.py

```diff
@@ -1,21 +1,22 @@
-from io import BytesIO
+from io import BytesIO, SEEK_SET, SEEK_END
 from threading import Lock
-
-from codalab.worker.un_gzip_stream import BytesBuffer
-
-
+import time
 class MultiReaderFileStream(BytesIO):
     """
-    FileStream that support multiple readers
+    FileStream that support multiple readers and seeks backwards
```

Comment:
Could you flesh this out into a much longer description (like what you have in the PR description), talking about what is maintained and what operations are supported, and what assumptions are made.

Reply:
Say things like: there are N readers, each reader has a current position...  buffer keeps bytes ... min reader position ... max reader position